chain/actions/src/errors.rs

```diff
@@ -57,14 +54,11 @@ pub enum ChainActionsError {
     #[error("indexer expectation has been unregistered")]
     ExpectationUnregistered,
 
-    #[error("no channel domain_separator tag found")]
-    MissingDomainSeparator,
-
-    #[error(transparent)]
-    DbBackendError(#[from] hopr_db_sql::errors::DbSqlError),
+    #[error("node db error: {0}")]
+    NodeDbError(Box<dyn std::error::Error + Send + Sync>),
 
     #[error(transparent)]
-    DbError(#[from] hopr_db_sql::api::errors::DbError),
+    IndexDbError(#[from] hopr_db_sql::errors::DbSqlError),
```

Comment:
Does this also solve the DbError and DbSqlError unification?

Reply:
Yes, there's no general DB API error, users of `hopr-api` must supply their own.
And `DbSqlError` will be gone with Indexer DB in 4.0.